js/app/widgets/themeableImage.js

```diff
@@ -185,47 +206,45 @@ var ThemeableImage = new Knowledge.Class({
         // a patch to fix this in Gtk if the need arises
     },
 
-    _draw_scaled_pixbuf: function (cr) {
-        // Helps to read these transforms in reverse. We center the pixbuf at
-        // the origin, scale it to fit, then translate its center to the
-        // center of our allocation.
-        cr.translate(this._image_width / 2, this._image_height / 2);
-        let scale = Math.min(this._image_width / this._pixbuf.get_width(),
-                             this._image_height / this._pixbuf.get_height());
-        scale = Math.min(1, scale);
-        cr.scale(scale, scale);
-        cr.translate(-this._pixbuf.get_width() / 2, -this._pixbuf.get_height() / 2);
-
-        Gdk.cairo_set_source_pixbuf(cr, this._pixbuf, 0, 0);
-        cr.paint();
-    },
-
     vfunc_draw: function (cr) {
-        let width = this.get_allocated_width();
-        let height = this.get_allocated_height();
-        Gtk.render_background(this.get_style_context(), cr,
-            0, 0, width, height);
-        Gtk.render_frame(this.get_style_context(), cr,
-            0, 0, width, height);
-        Gtk.render_focus(this.get_style_context(), cr,
-            0, 0, width, height);
+        let context = this.get_style_context();
+        let alloc_width = this.get_allocated_width();
+        let alloc_height = this.get_allocated_height();
         let padding = this._get_padding();
         let border = this._get_border();
 
-        width = width - padding.left - padding.right - border.left - border.right;
-        height = height - padding.top - padding.bottom - border.top - border.bottom;
-        if (this._pixbuf) {
-            let do_invalidate = this._image_width !== width || this._image_height !== height;
-            this._image_width = width;
-            this._image_height = height;
-            if (do_invalidate)
-                this._surface_cache.invalidate();
-            cr.setSourceSurface(this._surface_cache.get_surface(),
-                                padding.left + border.left, padding.top + border.top);
-            cr.paint();
-        } else {
-            Gtk.render_activity(this.get_style_context(), cr,
-                padding.left + border.left, padding.top + border.top,
+        let width = alloc_width - padding.left - padding.right - border.left - border.right;
+        let height = alloc_height - padding.top - padding.bottom - border.top - border.bottom;
+        if (this._pixbuf_width) {
+            let clamp = (x, low, high) => (((x) > (high)) ? (high) : (((x) < (low)) ? (low) : (x)));
```

Comment:
extra parens not needed

Reply:
Also, slightly more idiomatic would be
```js
function clamp(x, low, high) {
    return x > high ? high : x < low ? low : x;
}
```
(in the future we will have `Math.clamp()`, but it is still a proposal.)